packages/core/src/interfaces.ts

```diff
@@ -3,26 +3,52 @@ export interface ValidationOptions {
     runSerially?: boolean;
 }
 
-export interface BasicValidation {
-    module: string;
-    config?: any;
-    type: 'single' | 'bulk';
-    transformation?: any;
+export type ConstraintConfiguration = Partial<Record<string,any>> & {
+    __prerequisites?: ConstraintsMapping
 }
 
-export interface Validation extends BasicValidation {
-    description: string;
-    condition?: BasicValidation;
+export type ConstraintValue = boolean | string | number | ConstraintConfiguration;
+
+export interface ConstraintsMapping {
+    [path: string] : {
+        [constraintName: string] : ConstraintValue | [ConstraintValue],
+    } & {
+        __and?: (string | ConstraintsMapping)[]
```

Comment:
how this could be a string?

Reply:
really this special case should be omitted if we have an `and` module